libc/test/src/stdlib/quick_sort_test.cpp

```diff
@@ -1,16 +0,0 @@
-//===-- Unittests for quick sort ------------------------------------------===//
```

Comment:
this is important to keep for when heap sort is configured as the default sorting algorithm. We want to make sure we're testing all of the options.

Reply:
Genuine question, can you describe a scenario where not having this test would miss an actual issue? Presumably it's not always going to be configured as the default right?

Reply:
We want to test every sorting implementation regardless of what's the default. This file ensures we're testing the quick sort implementation, just like the `heap_sort_test` file ensures we're testing the heap sort implementation. 

A scenario where not having this test would miss an issue would be on a platform where `LIBC_QSORT_IMPL == LIBC_QSORT_HEAP_SORT` (e.g. the baremetal targets). At that point both the `qsort_test` and `heap_sort_test` are checking heapsort, with no testing for quick sort. Someone developing for that platform might make changes to the shared sorting code, but miss any breakages in quick sort.

Reply:
I had assumed both defines are always tested in the CI, by doing testing it this way we test the impl but not if actual user interface of setting a compile time define. I'll add the quick_sort test back in.

Reply:
Since we are not testing the public interface anyway, I've changed it to test quick_sort and heap_sort explicitly with as little code repeated as sensible in the tests.